lib/network/KMSInterface.ts

```diff
@@ -0,0 +1,68 @@
+export enum KmsType {
+    /** Internal scality provider */
+    int = 'int',
+    /** External provider */
+    ext = 'ext',
+};
+
+export enum KmsProtocol {
+    /** (SafeNet from Gemalto/Thales) */
+    scality = 'scality',
+    /** For tests & dev */
+    mem = 'mem',
+    /** Default if no provider */
+    file = 'file',
+    aws_kms = 'aws_kms',
+    kmip = 'kmip',
+};
+
+interface KmsTypeProtocolMapping {
+    [KmsType.int]: KmsProtocol.scality | KmsProtocol.mem | KmsProtocol.file,
+    [KmsType.ext]: KmsProtocol.aws_kms | KmsProtocol.kmip,
+};
+
+/**
+ * Possible arn are:
+ * 
+ * - `arn:scality:kms:int:scality:safenet:key/$keyId` (deprecated provider)
```

Comment:
I consider the deprecated `Safenet / Gemalto/Thales` using the `scality-kms` backend as internal provider with a protocol = `scality` and provider name = `safenet`.

Is this ok ?

Reply:
After discussion with david, this one should rather be an external

Reply:
Fixed https://github.com/scality/Arsenal/pull/2349/files#diff-1c24d37a19bc08405b548ef9b5ac12a2ef83b7edb3a2ee62cc8426c7df8e5b95R23-R26